src/lib/calculation.ts

```diff
@@ -136,74 +149,80 @@ export function calculate(input: TaxFormUserInput): TaxForm {
-                  (1 / 12) *
-                  this.r032_partner_pocet_mesiacov,
-              )
-        return round2decimal(result)
+        return this.r072_pred_znizenim.gt(36256.38)
+          ? Decimal.max(
+              0,
+              new Decimal(13001.438)
+                .minus(
+                  this.r072_pred_znizenim
+                    .times(1 / 4)
+                    .minus(Decimal.max(this.r032_partner_vlastne_prijmy, 0)),
+                )
+                .times(1 / 12)
+                .times(this.r032_partner_pocet_mesiacov),
+            )
+          : Decimal.max(
+              0,
+              new Decimal(3937.35).minus(
+                Decimal.max(this.r032_partner_vlastne_prijmy, 0)
+                  .times(1 / 12)
+                  .times(this.r032_partner_pocet_mesiacov),
+              ),
+            )
       }
-      return 0
+      return new Decimal(0)
     },
     get r076_kupele_spolu() {
-      return round2decimal(
-        this.r076a_kupele_danovnik + this.r076b_kupele_partner_a_deti,
-      )
+      return this.r076b_kupele_partner_a_deti.plus(this.r076a_kupele_danovnik)
     },
     get r076a_kupele_danovnik() {
-      return round2decimal(parseInputNumber(input?.r076a_kupele_danovnik ?? '0'))
+      return new Decimal(parseInputNumber(input?.r076a_kupele_danovnik ?? '0'))
     },
     get r076b_kupele_partner_a_deti() {
-      return round2decimal(
-        this.r033_partner_kupele_uhrady + this.r036_deti_kupele,
-      )
+      return this.r033_partner_kupele_uhrady.plus(this.r036_deti_kupele)
     },
     get r077_nezdanitelna_cast() {
-      return round2decimal(
-        Math.min(
-          this.r073 +
-            this.r074_znizenie_partner +
-            this.r075_zaplatene_prispevky_na_dochodok +
-            this.r076_kupele_spolu,
-          this.r072_pred_znizenim,
-        ),
+      return Decimal.min(
+        this.r073
+          .plus(this.r074_znizenie_partner)
+          .plus(this.r075_zaplatene_prispevky_na_dochodok)
+          .plus(this.r076_kupele_spolu),
+        this.r072_pred_znizenim,
       )
     },
     get r078_zaklad_dane_z_prijmov() {
-      return round2decimal(
-        Math.max(this.r072_pred_znizenim - this.r077_nezdanitelna_cast, 0),
+      return Decimal.max(
+        this.r072_pred_znizenim.minus(this.r077_nezdanitelna_cast),
+        0,
       )
     },
     get r080_zaklad_dane_celkovo() {
-      return round2decimal(floor(this.r078_zaklad_dane_z_prijmov, 2)) // TODO + tf.r065 + tf.r071 + tf.r079)
+      return this.r078_zaklad_dane_z_prijmov
+
+      /**TODO IMPORTANT Ked zaokruhlim centy nadol
+       * (https://podpora.financnasprava.sk/840887-Zaokr%C3%BAh%C4%BEovanie-platieb-zo-a-do-%C5%A1t%C3%A1tneho-rozpo%C4%8Dtu)
+       * Dostavam error o jeden cent. Ak nezaokruhlim, tak je to spravne a
+       aj eformom. Zeby mali oni bug?
+       */
+      // return floorDecimal(this.r078_zaklad_dane_z_prijmov) // TODO + tf.r065 + tf.r071 + tf.r079)
```

Comment:
Checkni toto @mathio. Ked to zaokruhlim na eurocenty nadol, tak ako je to v pouceni, tak dostavam chybu o jeden cent (aj som doublecheckoval ci testy su spravne a su, aj formular to hlasi ako chybu)

Existuje na to nejake vysvetlenie alebo naozaj maju bug vo formulari a odrbavaju ludi o jednocentovky? :D

Reply:
"Na riadku 80 daňovník vypočíta základ dane v súlade s § 4 zákona (r. 78 + 65 + 71+ 79), pričom takto vypočítaný základ dane zistený podľa § 4 zákona zvýši aj o sumu uvedenú na riadku 79. Daň podľa § 15 písm. a) bodu 1 zákona daňovník vypočíta na riadku 81 zo základu dane zisteného podľa § 4 zákona zaokrúhleného na eurocenty nadol, pričom aj daň sa zaokrúhľuje na eurocenty nadol."

Reply:
Alebo mozno zle citam to poucenie, nie je to napisane uplne jasne

Reply:
Neviem 🤷‍♂️ Navrhujem spytat sa poradkyn na konkretny priklad (konkretne cisla) kde ti to takto vychadza. Ak by odrbavali o cent tak by sa na to uz davno prislo, vela softverov generuje danove :)

Reply:
Nasiel som par miest kde nepouzivas decimal, mozno to ovplyvnuju tie?

Reply:
Asi to necham takto, pokial cisla vychadzaju spravne tak to asi tak ma byt 🤷‍♂️